src/components/Swap/Summary/index.tsx

```diff
@@ -19,6 +22,75 @@ import Summary from './Summary'
 
 export default Summary
 
+enum ReviewState {
+  REVIEWING,
+  ALLOWANCE_PENDING,
+  ALLOWANCE_FAILED,
+  TRADE_CHANGED,
+  SWAP_PENDING,
+}
+
+function useReviewState(
+  trade: InterfaceTrade,
+  onConfirm: () => Promise<void>,
+  allowance: Allowance,
+  doesTradeDiffer: boolean
+) {
+  const [currentState, setCurrentState] = useState(ReviewState.REVIEWING)
+  const { onSubmitSwapClick } = useAtomValue(swapEventHandlersAtom)
+
+  useEffect(() => {
+    if (doesTradeDiffer && currentState === ReviewState.REVIEWING) {
+      setCurrentState(ReviewState.TRADE_CHANGED)
+    } else if (!doesTradeDiffer && currentState === ReviewState.TRADE_CHANGED) {
+      setCurrentState(ReviewState.REVIEWING)
+    }
+  }, [currentState, doesTradeDiffer])
+
+  const triggerSwap = useCallback(async () => {
+    setCurrentState(ReviewState.SWAP_PENDING)
+    onSubmitSwapClick?.(trade)
+    await onConfirm()
+    setCurrentState(ReviewState.REVIEWING)
+  }, [onConfirm, onSubmitSwapClick, trade])
+
+  const prevAllowanceRef = useRef(allowance)
+  // Automatically triggers signing swap tx if allowance requirements are met
+  useEffect(() => {
+    if (prevAllowanceRef.current.state === AllowanceState.REQUIRED && allowance.state === AllowanceState.ALLOWED) {
+      // Prevents swap if trade has updated mid permit2 flow
+      if (doesTradeDiffer) setCurrentState(ReviewState.TRADE_CHANGED)
+      else triggerSwap()
```

Comment:
whoa the if/else without brackets is spooky 🤔 but i dont hate it lol

Reply:
before I would've done `(doesTradeDiffer) ? setCurrentState(ReviewState.TRADE_CHANGED) : triggerSwap()` but you mentioned that means jail

Reply:
ok yes slight jail for that